src/tree.rs

```diff
@@ -827,6 +1005,87 @@ enum Divergence {
     Diverged,
 }
 
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum Problem {
+    /// The item doesn't have a `parentid`, and isn't mentioned in any
+    /// folder's `children`.
+    Orphan,
```

Comment:
I'm using "orphan" to mean "doesn't have a parent at all, we have no idea where to put it", not "has a `parentid` that doesn't exist, so we can't put it there"—but maybe that distinction isn't useful?